### DISCUSSION TITLE
[Access to nodal variable value][Postprocessor]

### DISCUSSION THREAD

**Discussion by coskrrb2002** [idaholab/moose#17952] [Q&A General] [answered]
Dear users,

I just found a way to access nodal variable values in postprocessors like the below (bold).

But this method only can access individual nodal id.

Is there any way to call all nodal variable values in postprocessors?

Or how can I load a number of nodes in mesh?
Then, maybe I can use iterations to call all the variable values.

I have looked through the whole postprocessors in moose but I could not find anything.

---------------------------------------------------------------------------
Real
NodalVariableValue::getValue()
{
  Real value = 0;

  if (_node_ptr && _node_ptr->processor_id() == processor_id())
    _**value = _subproblem.getStandardVariable(_tid, var_name).getNodalValue(*_node_ptr);**_

  gatherSum(value);

  return _scale_factor * value;
}
---------------------------------------------------------------------------

**Comment by loganharbour** [comment]
Can you describe specifically what you are trying to accomplish? That is, what are you trying to output and when?

**Reply by coskrrb2002** [reply]
Dear loganharbour,


I am trying to make a precipitation of U(VI) in my transient model.

So, if U(VI) is reached certain threshold concentration, it will be precipitated via kinetic constant with U(VI) complex.

To make this, I need a postprocess every end of time step and this precipiated U(VI) complex should be an accumulated value.

Hence, I would like to access variable value end of each time step for whole nodal value to get concentration of U(VI) and judge whether it is saturated and start to precipitate.

Then, I want to make the accumulation of U(VI) complex.

**Comment by coskrrb2002** [comment, marked as answer]
Oh, I found the way to do this and made it!

I used the following postprocessors and auxkernel.

CumulativeValuePostprocessor, PostprocessorAux, and I modified NodalExtremeValue to get nodal value.

So, Modified NodalExtremeValue load variable values in nodes, then the CumulativeValuePostprocessor calculated the accumulation of nodal variable values.

After that PostrpocessorAux assigned accumulated values into auxvariables!

Thank you so much loganharbour!
